Sources/Sentry/SentryBinaryImageCache.m

```diff
@@ -83,6 +88,16 @@ - (void)binaryImageAdded:(const SentryCrashBinaryImage *)image
 
         [_cache insertObject:newImage atIndex:left];
     }
+
+    if (self.isDebug) {
+        // Only validate if debug is enabled
```

Comment:
`l`: Maybe we should mention this code adds a bit of of overhead to explain why we only do it in debug